modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/sql/JdbcQueryCursor.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.handler.requests.sql;
+
+import java.util.Iterator;
+import org.apache.ignite.internal.sql.engine.ResultSetMetadata;
+import org.apache.ignite.internal.sql.engine.SqlCursor;
+import org.apache.ignite.internal.sql.engine.SqlQueryType;
+
+/**
+ * Sql query cursor with jdbc related parameters.
+ */
+public class JdbcQueryCursor<T> implements SqlCursor<T> {
+    /** Max rows. */
+    private final long maxRows;
+
+    /** Query result rows. */
+    private final SqlCursor<T> cur;
+
+    /** Query results iterator. */
+    private final Iterator<T> iter;
```

Comment:
why do you need an iterator? SqlCursor implements Iterator, so you could use it directly